### PR TITLE
[DOP-19796] - add log_url before sending run task to celery worker

### PR DESCRIPTION
**Pull request by maxim-lixakov**
## Change Summary

- Add logic for updating `log_url` in `Run` table rendered from template for future viewing of celery worker logs
- Add `WorkerSettings` to configure work of celery worker (logging, log_url_template, etc.)

## Checklist

* [x] Commit message and PR title is comprehensive
* [x] Keep the change as small as possible
* [x] Unit and integration tests for the changes exist
* [x] Tests pass on CI and coverage does not decrease
* [ ] Documentation reflects the changes where applicable
* [ ] `docs/changelog/next_release/<pull request or issue id>.<change type>.rst` file added describing change
  (see [CONTRIBUTING.rst](https://github.com/MobileTeleSystems/syncmaster/blob/develop/CONTRIBUTING.rst) for details.)
* [x] My PR is ready to review.

### CODE REVIEW

File: syncmaster/utils.py [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+
+
+from jinja2 import Template
+
+from syncmaster.config import Settings
+
+settings = Settings()
+
+
+def render_log_url(run, correlation_id):
```
Comment (IlyasDevelopment) [inline syncmaster/utils.py:12]: nit: можно добавить type hints

**Review by IlyasDevelopment — APPROVED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -52,6 +52,7 @@ pydantic-settings = { version = "^2.5.2", optional = true }
 sqlalchemy = { version = "^2.0.35", optional = true }
 sqlalchemy-utils = { version = "^0.41.2", optional = true }
 fastapi = { version = "^0.115", optional = true}
+asgi-correlation-id = {version = ">=4.2.0", optional = true, python = ">=3.8"}
```
Comment (dolfinus) [inline pyproject.toml:55]: BTW, why adding python contraing while python version is already fixed `python = "^3.11"`

File: syncmaster/config.py [hunk 3]
```diff
@@ -36,6 +36,25 @@ class Settings(BaseSettings):
...
+    #  TODO: implement jinja universal config
+    LOG_URL_TEMPLATE: str = """
+        {% if logging_system == 'elk' %}
+        {{ kibana_host }}?correlation_id={{ correlation_id }}&run_id={{ run.id }}
+        {% elif logging_system == 'grafana' %}
+        {{ grafana_host }}?correlation_id={{ correlation_id }}&run_id={{ run.id }}
+        {% else %}
+        Unsupported logging system: {{ logging_system }}
+        {% endif %}
+        """
```
Comment (dolfinus) [inline syncmaster/config.py:56]: Why not allowing admin to set up Jinja template for URL?

File: syncmaster/backend/api/v1/runs.py [hunk 4]
```diff
@@ -104,6 +107,10 @@ async def start_run(
         transfer.target_connection_id,
     )
 
+    request_correlation_id = correlation_id.get()
+    if not request_correlation_id:
+        request_correlation_id = str(uuid.uuid4())
```
Comment (dolfinus) [inline syncmaster/backend/api/v1/runs.py:112]: How does this value is being passed to Celery worker?

File: syncmaster/config.py (hunk 3 above)
Comment (maxim-lixakov) [reply syncmaster/config.py:56]: where should admin set this template?

Comment (dolfinus) [reply syncmaster/config.py:56]: Using environment variables, just like other settings, like DB url

File: syncmaster/backend/api/v1/runs.py (hunk 4 above)
Comment (maxim-lixakov) [reply syncmaster/backend/api/v1/runs.py]: added logging settings for worker where correlation is formatted into log message

File: syncmaster/settings/__init__.py [hunk 5]
```diff
@@ -47,6 +48,7 @@ class Settings(BaseSettings):
     SECRET_KEY: str = "secret"
     SECURITY_ALGORITHM: str = "HS256"
     CRYPTO_KEY: str = "UBgPTioFrtH2unlC4XFDiGf5sYfzbdSf_VgiUSaQc94="
+    LOG_URL_TEMPLATE: str = ""
```
Comment (dolfinus) [inline syncmaster/settings/__init__.py:51]: This should be worker-only option.

File: tests/test_unit/test_runs/test_create_run.py [hunk 6]
```diff
@@ -151,6 +154,8 @@ async def test_superuser_can_create_run(
         "transfer_dump": run.transfer_dump,
     }
     assert result.status_code == 200
+    assert "correlation_id" in result.json().get("log_url")
+    assert "run_id" in result.json().get("log_url")
```
Comment (dolfinus) [inline tests/test_unit/test_runs/test_create_run.py:158]: Assign `result.json()` to variable

File: syncmaster/settings/__init__.py (hunk 5 above)
Comment (dolfinus) [reply syncmaster/settings/__init__.py:51]: Comment is resolved, but these options are still in the root Settings class

**Review by dolfinus — APPROVED** [review]

**Pull request merged**